File: server/src/game/tools/conf/output/ts/SuitConfig.ts
class SuitConfig {
    private static _dic:Dictionary<SuitConfig> = new Dictionary<SuitConfig>();
    private static _arr:Array<SuitConfig> = new Array<SuitConfig>();

	 public Id :number ;
     public Name :string ;
     public Parts : Array<number>  =new Array<number>();
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<SuitConfig>{
        return  SuitConfig._dic;
    }
    public static get Arr():Array<SuitConfig>{
        return SuitConfig._arr;
    }
    public static parse(json:any[]){

        for(var i = 0; i < json.length; i++){
            let config:SuitConfig = <SuitConfig>json[i];
            SuitConfig._arr.push(config);
            SuitConfig._dic.add(config.Id,config);
        }
    }
}

File: server/src/game/conf/SuitConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type SuitConfigElem struct {
	Id int
	Name string
	Parts []int
	
}

type SuitConfig0 struct {
	Dic map[int]*SuitConfigElem
	Arr []*SuitConfigElem
}

var (
	SuitConfig SuitConfig0
)

func loadSuitConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	SuitConfig.Dic = make(map[int]*SuitConfigElem)
	SuitConfig.Arr = make([]*SuitConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &SuitConfig.Arr)
	for _,conf := range SuitConfig.Arr{
		SuitConfig.Dic[conf.Id] = conf
	}
	
}
